keras/backend/tensorflow_backend.py

```diff
@@ -427,7 +427,8 @@ def rnn(step_function, inputs, output_dim, initial_states,
         new_states: list of tensors, latest states returned by
             the step function, of shape (samples, ...).
     '''
-    inputs = tf.transpose(inputs, (1, 0, 2))
+    axes = [1, 0] + range(2, len(inputs.get_shape()))
```

Comment:
I think it would be good to assert than the input tensor is at least 3D (to avoid a more obscure error message later).